### PR TITLE
[Bugfix] Fix feature size calculation for LLaVA-NeXT

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR primarily updates our code to be consistent with https://github.com/huggingface/transformers/pull/32314.

The feature size calculation has also been updated to no longer depend on the floating-point error in CUDA for some specific image resolutions, e.g.:

```py
# original_height = 183, original_width = 488, height = 24
>>> int(183 * torch.tensor(24).cuda() / 488)
8
>>> int(183 * torch.tensor(24).cpu() / 488)
9
```

This fixes a bug in which the number of image placeholder tokens becomes incorrect for such edge cases when the model is run on other devices such as CPU.

cc @xwjiang2010

### CODE REVIEW

File: vllm/model_executor/models/internvl.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 
 class InternVLImagePixelInputs(TypedDict):
     type: Literal["pixel_values"]
-    data: BatchedTensors
+    data: Union[torch.Tensor, List[torch.Tensor]]
```
Comment (ywang96) [inline vllm/model_executor/models/internvl.py:46]: Why do we no longer use `BatchedTensors`?

**Review by ywang96 — APPROVED** [review]
LGTM but should we wait for a new `transformers` release then merge this with a version bump? It looks like that PR hasn't been released yet.

**Comment by DarkLight1337** [comment]
> LGTM but should we wait for a new `transformers` release then merge this with a version bump? It looks like that PR hasn't been released yet.

It's an internal change so it shouldn't break anything. I think we can do this in parallel with `transformers` development.

**Comment by ywang96** [comment]
> > LGTM but should we wait for a new `transformers` release then merge this with a version bump? It looks like that PR hasn't been released yet.
> 
> It's an internal change so it shouldn't break anything. I think we can do this in parallel with `transformers` development.

I'm only worried about the model result consistency, but I guess since we're developing on the main branch it shouldn't be a big deal.

**Comment by DarkLight1337** [comment]
> > > LGTM but should we wait for a new `transformers` release then merge this with a version bump? It looks like that PR hasn't been released yet.
> > 
> > 
> > It's an internal change so it shouldn't break anything. I think we can do this in parallel with `transformers` development.
> 
> I'm only worried about the model result consistency, but I guess since we're developing on the main branch it shouldn't be a big deal.

Yeah, either way there will be some inconsistency as we can't time our release to be at the same time as `transformers`.

**Review by xwjiang2010 — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**